### PR TITLE
DEVDOCS-6104: Clarify the meaning of version field in the request/response of cart/checkout API

### PR DESCRIPTION
**Pull request by davidchin**
# [DEVDOCS-6104]

## What changed?
* Clarify the meaning of `version` field in the request/response of cart/checkout API so developers know what it can be used for.

## Release notes draft
N/A

ping @bigcommerce/team-checkout 

[DEVDOCS-6104]: https://bigcommercecloud.atlassian.net/browse/DEVDOCS-6104?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by animesh1987 — DISMISSED** [review]

**Review by valentindellangela — APPROVED** [review]

File: reference/carts.sf.yml [hunk 1]
```diff
@@ -402,7 +402,7 @@ components:
           description: Locale of the cart.
         version:
           type: integer
-          description: The current version of the cart.
+          description: The current version of the cart, which increments with each successful update. It can be used to enable optimistic concurrency control for subsequent updates.
```
Comment (bc-traciporter) [inline reference/carts.sf.yml:405]:
```suggestion
          description: The current version of the cart increments with each successful update. You can use it to enable optimistic concurrency control for subsequent updates.
```

File: reference/carts.v3.yml [hunk 2]
```diff
@@ -1770,7 +1770,7 @@ components:
                   type: string
         version:
           type: integer
-          description: The current version of the cart.
+          description: The current version of the cart, which increments with each successful update. It can be used to enable optimistic concurrency control for subsequent updates.
```
Comment (bc-traciporter) [inline reference/carts.v3.yml:1773]:
```suggestion
          description: The current version of the cart increments with each successful update. You can use it to enable optimistic concurrency control for subsequent updates.
```

File: reference/checkouts.sf.yml [hunk 3]
```diff
@@ -2343,7 +2343,7 @@ components:
             `true` value indicates StoreCredit has been applied.
         version:
           type: integer
-          description: The current version of the checkout.
+          description: The current version of the checkout, which increments with each successful update. It can be used to enable optimistic concurrency control for subsequent updates.
```
Comment (bc-traciporter) [inline reference/checkouts.sf.yml:2346]:
```suggestion
          description: The current version of the checkout increments with each successful update. You can use it to enable optimistic concurrency control for subsequent updates.
```

File: reference/checkouts.v3.yml [hunk 4]
```diff
@@ -8727,7 +8727,7 @@ components:
                       description: Text of the banner.
         version:
           type: integer
-          description: The current version of the checkout.
+          description: The current version of the checkout, which increments with each successful update. It can be used to enable optimistic concurrency control for subsequent updates.
```
Comment (bc-traciporter) [inline reference/checkouts.v3.yml:8730]:
```suggestion
          description: The current version of the checkout increments with each successful update. You can use it to enable optimistic concurrency control for subsequent updates.
```

**Review by bc-traciporter — COMMENTED** [review]
Hi @Lucki2501 I made some editorial updates. I removed passive voice. Let me know if you think the changes are okay or if they change the meaning.

**Review by bc-shawnwang — COMMENTED** [review]
it's good on the words itself, I'm just wondering if there is the way to extract this general field and every place just refer it.

**Comment by davidchin** [comment]
@bc-shawnwang Yeah I agree, I also notice there's a lot of duplication. I'll leave it to the dev-docs team to restructure the yaml files.

**Comment by bc-traciporter** [comment]
Thank you @bc-shawnwang, @valentindellangela and @animesh1987 . I would like to close this PR for now and think about how to better restructure this file to remove the duplicates. I wasn't able to figure out how to do it quickly. I will need to think about it.

File: reference/carts.sf.yml [hunk 5]
```diff
@@ -537,7 +537,7 @@ components:
                 $ref: '#/components/schemas/requestCartPostLineItem'
             version:
               type: integer
-              description: The expected version of the cart.
+              description: The cart version to which updates are expected to apply. If the provided version doesn't match the current cart version, a conflict error will be returned. This field is optional; if not provided, optimistic concurrency control will not apply.
```
Comment (bc-andreadao) [inline reference/carts.sf.yml:540]:
```suggestion
              description: The cart version that you expect to apply the updates. If the provided version doesn't match the current cart version, you will receive a conflict error. This field is optional; if not provided, optimistic concurrency control will not apply.
```

**Comment by bc-andreadao** [comment]
@bc-traciporter My [PR 468](https://github.com/bigcommerce/docs/pull/468) restructures the storefront checkout yml file, so that endpoint would be taken cared of

**Comment by bc-traciporter** [comment]
> @bc-traciporter My [PR 468](https://github.com/bigcommerce/docs/pull/468) restructures the storefront checkout yml file, so that endpoint would be taken cared of

Awesome @bc-andreadao

File: reference/carts.sf.yml (hunk 5 above)
Comment (bc-traciporter) [reply reference/carts.sf.yml]: I will update the other occurrences. @bc-andreadao

**Review by bc-andreadao — APPROVED** [review]

**Pull request merged**